woodwork/schema.py

```diff
@@ -219,8 +221,15 @@ def _filter_cols(self, include):
                     continue
                 elif selector in tags_in_schema:
                     tags_used.add(selector)
+                elif col_names and selector in self.columns:
+                    cols_to_include.add(selector)
```

Comment:
@thehomebrewnerd Thought about why column order would be last in ltype > semantic tag > column name preference, and I think it's because there can only ever be one column with that column name, but there can many columns with a certain ltype or tag.

That said, it's not ideal that, if you have a column named `full_name` but its logical type is `NaturalLanguage`, it'll never get selected for even if you put `full_name` in the include list. But if we have it come first, then any column's with that logical type but not that column name will not get included.

My suggestion is that if a selector is a column name, it gets included. And then after that, we go and check logical types and semantic tags.

Reply:
Wondering if the root of the issue is that we have a method that isn't doing just one thing, it's potentially doing multiple things depending on the input? Not suggesting we do this now, but maybe we need to revisit this and refactor at some point so this behavior is separated more cleanly somehow.

Maybe for now this should be an AND type selection, such that if there were three columns, one with the name `full_name`, one with the LogicalType `FullName` and one with the semantic tag of `full_name` all three would get selected with a `full_name` selector. Then the burden is on the user to avoid name conflicts in their data, types and tags, and we don't have to make the decision about what should be returned in the case of collisions.

Reply:
Yeah, it reminds me how with `DataTable.set_types` we specifically have separate logical types and semantic tags parameters in order to not blur the lines between semantic tags and logical types.

For now, I think having an AND selection makes a lot of sense.

Reply:
@thehomebrewnerd Changed `_filter_cols` to add the selector if it matches any of logical type, column name, or semantic tag.

I'm not feeling great about how that code is looking at the moment, though, since it feels overly complicated/hard to understand. I'm going to spend some more time thinking about ways to simplify, but open to any suggestions!